javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -387,6 +408,26 @@ default int length() {
     <V2> Map<K, V2> mapValues(Function<? super V, ? extends V2> valueMapper);
 
     /**
+     * Calculates the <em>maximum</em> of this {@code Map} values using the natural {@code Comparator} of the values, if exists.
+     *
+     * @return {@code Some(maximum)} of this elements or {@code None} if this is empty or no {@code Comparator} is applicable
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    default Option<Tuple2<K, V>> max() {
+        if (isEmpty()) {
+            return Option.none();
+        } else {
+            if (head()._2 instanceof Comparator) {
```

Comment:
> Comparable?

The tests use `IntMap<T>` which extends `Traversable<T>`. We need to extends `Map` in order to test the overrides.

However, currently we need IntMap because our AbstractMapTest extends the AbstractTraversableTest and otherwise the method overrides do not compile.

I will search for another solution.

This PR: simple change, big impact o_O